crates/ruff_server/src/server/schedule.rs

```diff
@@ -42,13 +42,13 @@ pub(crate) struct Scheduler {
 impl Scheduler {
     pub(super) fn new(
         session: Session,
-        thread_count: usize,
+        pool_threads: usize,
         sender: &Sender<lsp_server::Message>,
     ) -> Self {
         Self {
             session,
             fmt_pool: thread::Pool::new(1),
-            background_pool: thread::Pool::new(thread_count),
+            background_pool: thread::Pool::new(pool_threads),
```

Comment:
I think we should change `Pool::new` to take a `NonZeroUsize` (and pass it as such all the way to the `::new` call) because the `Pool` is non-functioning when created with `0` (it has 0 threads, and the bounded queue has a capacity of 0, dropping all messages).

Not related to this PR but I noticed it when looking at the `Pool::new` definition. 

```
        let (job_sender, job_receiver) = crossbeam::channel::bounded(threads);
```

We should revisit the bounded queue's size and how a queue overflow is handled (currently panics). I'm saying that because using `--threads=1` created a bounded queue of size 1, and the server panics as soon as we queue the second message, which doesn't seem unlikely. I prefer a fixed message count that we allow to accumulate regardless of the thread count.

Reply:
I agree with this, though I have a question:
> We should revisit the bounded queue's size and how a queue overflow is handled (currently panics).

Is this true? The [documentation](https://docs.rs/crossbeam/latest/crossbeam/channel/fn.bounded.html) says that `send`-ing a message when the queue is at capacity will block the thread until space becomes available.

Reply:
Oh good point. I misunderstood the API. I guess a too small bounded queue is less of a concern in that case (do you know if it would be possible to lead to dead-locks when setting the count to 1 and the server sends a notification that requires a response?)

Reply:
> (do you know if it would be possible to lead to dead-locks when setting the count to 1 and the server sends a notification that requires a response?)

No, because any tasks active in the thread pool are not allowed to await server connections. They can _send_ requests/notifications to the server, but they can't receive messages. The sender is one half of a zero-sized channel onto a dedicated I/O thread, and that shouldn't block unless the I/O between the server and client itself starts blocking.